md-dao/src/main/java/com/wipro/ats/bdre/md/dao/GetDotStringDAO.java

```diff
@@ -53,30 +56,30 @@
             List<String> srcTargetNodeIdList = new ArrayList<String>();
 
             //List of sub processes of parent process passed
-            Criteria fetchListOfSubProcess = session.createCriteria(com.wipro.ats.bdre.md.dao.jpa.Process.class).add(Restrictions.eq("process.processId", processId)).setProjection(Projections.property("processId"));
+            Criteria fetchListOfSubProcess = session.createCriteria(com.wipro.ats.bdre.md.dao.jpa.Process.class).add(Restrictions.eq("process.processId", processId)).setProjection(Projections.property(PROCESSID));
             LOGGER.info("Number of sub processes:" + fetchListOfSubProcess.list().size());
             List<Integer> subProcessList = fetchListOfSubProcess.list();
 
             // select max(instance_exec_id) from lineage_query group by process_id having process_id in
-            if (fetchListOfSubProcess.list().size() != 0) {
-                Criteria fetchMaxIeId = session.createCriteria(LineageQuery.class).setProjection(Projections.groupProperty("processId")).add(Restrictions.in("processId", subProcessList)).setProjection(Projections.max("instanceExecId"));
+            if (!fetchListOfSubProcess.list().isEmpty()) {
+                Criteria fetchMaxIeId = session.createCriteria(LineageQuery.class).setProjection(Projections.groupProperty(PROCESSID)).add(Restrictions.in(PROCESSID, subProcessList)).setProjection(Projections.max("instanceExecId"));
                 List<Long> maxIeIdList = fetchMaxIeId.list();
                 LOGGER.info("max instance exec Id list size:" + fetchMaxIeId.list().size());
                 //select query_id from lineage_query where instance_exec_id in
 
-                if (fetchMaxIeId.list().size() != 0) {
+                if (!fetchMaxIeId.list().isEmpty()) {
                     Criteria fetchQueryId = session.createCriteria(LineageQuery.class).add(Restrictions.in("instanceExecId", maxIeIdList)).setProjection(Projections.property("queryId"));
                     LOGGER.info("Number of query ids fetched:" + fetchQueryId.list().size());
                     List<String> queryIdList = fetchQueryId.list();
 
-                    if (fetchQueryId.list().size() != 0) {
+                    if (!fetchQueryId.list().isEmpty()) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)